### PR TITLE
Support Azure DevOps CI

### LINKED ISSUE #8: [Template] Include options for build scripts for GHA and Azure DevOps
## What would you like to be added:

CI build scripts would
- build and package for each platform
- run any unit and/or ui tests

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renders for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks

### PR DESCRIPTION
**Pull request by dansiegel**
GitHub Issue (If applicable): closes #8

## PR Type

- Feature

## What is the current behavior?

Only GitHub Actions is supported

## What is the new behavior?

You can now get a basic Azure Pipelines build

### CODE REVIEW

File: src/Uno.Templates/content/unoapp/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  value: 'windows-2022'
+
+- name: macAgent
+  value: 'macOS-13'
+
+stages:
+  - stage: tests
+    displayName: Tests
+    jobs:
+      - template: build/jobs/smoke-test.yml
```
Comment (nickrandolph) [inline src/Uno.Templates/content/unoapp/azure-pipelines.yml:73]: For GHA we have both jobs inline in the ci.yml. Can we be consistent and either separate the GHA jobs into separate files, or move these jobs inline

File: src/Uno.Templates/content/unoapp/azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  default: 'Apple Development: Contoso Contoso (752234A22L)'    #The name of the distribution/development certificate
+#endif
+
+variables:
+- name: jobTimeout
+  value: '60'
+
+# For the latest available agents see:
+# https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml&&WT.mc_id=DT-MVP-5002924
+- name: windowsAgent
```
Comment (nickrandolph) [inline src/Uno.Templates/content/unoapp/azure-pipelines.yml:63]: GHA is running on windows-latest. Can we be consistent? Or are do they maintain different hosts?

File: src/Uno.Templates/content/unoapp/azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  value: 'windows-2022'
+
+- name: macAgent
```
Comment (nickrandolph) [inline src/Uno.Templates/content/unoapp/azure-pipelines.yml:66]: Let's remove variables that aren't used

File: src/Uno.Templates/content/unoapp/azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
+###################################################
+#                    IMPORTANT                    #
+# Please make sure  GitTools is installed in your #
+# project. Otherwise all jobs will fail           #
+# See how to install new extensions here:         #
+# https://learn.microsoft.com/en-us/azure/devops  #
```
Comment (nickrandolph) [inline src/Uno.Templates/content/unoapp/azure-pipelines.yml:6]: Let's remove this warning and anything else that's not used in the current pipeline

File: src/Uno.Templates/content/unoapp/azure-pipelines.yml (hunk 1 above)
Comment (dansiegel) [reply src/Uno.Templates/content/unoapp/azure-pipelines.yml:73]: will follow up with a separate PR for GHA

**Review by nickrandolph — APPROVED** [review]

**Pull request merged**